src/lua/LuaGame.cpp

```diff
@@ -316,7 +317,49 @@ static int l_game_save_game(lua_State *l)
 static int l_game_delete_save(lua_State *l)
 {
 	const std::string filename(luaL_checkstring(l, 1));
-	lua_pushboolean(l, Game::DeleteSave(filename));
+	lua_pushboolean(l, SaveGameManager::DeleteSave(filename));
+	return 1;
+}
+
+/*
+ * Function: ListSaves
+ *
+ * Retrieves a list of all game saves along with their last-modified time.
+ *
+ * > Game.ListSaves()
+ *
+ * Parameters:
+ *
+ * Return:
+ *
+ *   saves - a list of save-games as names and modification-date
+ *
+ * Availability:
+ *
+ *   September 2024
+ *
+ * Status:
+ *
+ *   experimental
+ */
+static int l_game_list_saves(lua_State *l)
+{
+	auto saves = SaveGameManager::ListSaves();
+
+	lua_newtable(l);
+	int savesTable = lua_gettop(l);
+	int saves_len = 0;
+
+	for (const auto &save : saves)
+	{
+		lua_newtable(l);
+		pi_lua_settable(l, "name", save.GetName().c_str());
+		pi_lua_push_date_time(l, save.GetModificationTime());
+		lua_setfield(l, -2, "mtime");
```

Comment:
Note that if the related comment about `pi_lua_generic_push` is implemented, this can use the LuaTable wrapper for convenience:

```c++
LuaTable(l) save;
save.Set("name", save.GetName());
save.Set("mtime", save.GetModificationTime());
```

Reply:
I'm away for the weekend; will pick this up next week. Thank you for the suggestions.

Reply:
Fixed (?) in e3fca64b76ea7b51d5f18b7e63a2c7d6cf486a55